pytorch_lightning/core/optimizer.py

```diff
@@ -162,3 +164,220 @@ def closure_dis():
         assert trainer is not None
         with trainer.profiler.profile(profiler_action):
             trainer.training_type_plugin.optimizer_step(self._optimizer, self._optimizer_idx, closure, **kwargs)
+
+
+def _init_optimizers_and_lr_schedulers(model: "pl.LightningModule") -> Tuple[List, List, List]:
```

Comment:
```suggestion
def init_optimizers_and_lr_schedulers(model: "pl.LightningModule") -> Tuple[List, List, List]:
```

let's add a short docstring here as well

Reply:
why should this be on the public API? do we expect end users to call it directly? @rohitgr7 previously wanted me to change this to private

Reply:
I see, to start with we could make it private, and make sure the API is tested & stable. Since it's meant to be called externally by the strategy implementations (not end users), I think eventually this should be public. But it's minor if we want to make it private first and later make it public. That's always easier than the other way around 

@awaelchli @rohitgr7 what do you think?

Reply:
from the user's point, it should be private, but as @ananthsub mentioned that this is being accessed outside its module. tbh we have already broken this thing at a few other places already (accessing private functions outside their modules), so not sure if it's a common thing in FOSS. IMO keep it private.

Reply:
Public or protected, fine with me. Private is a bit extreme considering it's getting imported/called from outside this module.

Reply:
I think we should keep it private since it's not meant to be called by end users, and as Rohit said, we already break the rule of accessing private functions outside their modules all over the codebase

Reply:
Oh, didn't realize private and protected are different things. Then we can make it protected (which is is now).